### DISCUSSION TITLE
Is it possible to programmatically add users and roles in superset?

### DISCUSSION THREAD

**Discussion by victarary** [apache/superset#15545] [Q&A / Help] [answered]
We have a demand that needs to add users and roles in our business system with superset APIs, we also need to assign users to roles, dashboards, charts and datasets programmatically. I can only see very few APIs about security in the following documentation. Is there any instruction how we can configure access control programmatically?
https://superset.apache.org/docs/rest-api

I'm a junior in python flask and superset, it's very appreciated, if anyone has fulfilled similar demands can provide a solution.

**Comment by algers** [comment, marked as answer]
Users, roles, permissions use the [Flask-Appbuilder scaffold](https://flask-appbuilder.readthedocs.io/en/latest/api.html)

**Reply by EuphoriaCelestial** [reply]
@algers can you kindly give me an example code? or more detailed guide?
I am new to Flask and Superset, I don't know what to do with Appbuilder

**Reply by maximStafeevAtAllscripts** [reply]
https://github.com/apache/superset/discussions/23745

**Reply by Rkejji** [reply]
I wrote [this article](https://medium.com/p/43f19473d57e) explaining how to handle permissions, roles and users programmatically in Superset, I hope it helps
